### ISSUE TITLE
Game presets

### ISSUE THREAD

**Issue by WandererXII** [WandererXII/lishogi#782]
As was discussed here: https://github.com/WandererXII/lishogi/pull/545.

Current 'modal view' (let's call it that) presented when creating a game can be rather confusing for new players, there is just too many options:
![image](https://github.com/WandererXII/lishogi/assets/43043181/478aa269-b8e9-4572-8cdd-81f70a3c340b)

This is trying to simplify the game creation and guide users towards fewer time controls.
I also used two buttons for creating correspondence game, and we have a whole row for games against computer - the middle one being the highest level you defeated so far (starting after the update). Individual time controls might change.
![image](https://github.com/WandererXII/lishogi/assets/43043181/7ab696fd-2f10-4982-b96f-5cd45b0d4440)

This is not quickplay the way lichess implements it, we couldn't feed it with the limited amount of players we have. This just creates a hook for you in the lobby, the same one you would get from using the modal view. So after clicking the preset, the tab switches to lobby tab, for correspondence games to correspondence tab and for ai games it just sends you to the game as you would expect. If a game with the same time control in the lobby already exists you will be auto joined.

Possible issues:

- Users don't know the time control where someone might already be waiting. They can see this only after selecting some time control and being redirected to the lobby tab. This might be solved by somehow having the time control highlighted or glowing when a hook with the same time control already exists. Something like this, but it seems a little cluttered:
![image](https://github.com/WandererXII/lishogi/assets/43043181/daee9108-2187-4f6e-acb5-ed1c86cefaeb)

- Ratings - We don't really want to auto join people with a large difference in ratings, but setting too narrow rating range would result in no matches whatsoever. For now I limited the hook to +-300 users rating. I would also like to show the rating range somewhere in the lobby.

**Comment by WandererXII** [comment]
About the first possible issue:

Having text there seems like too much information and makes it look not centered.
![alt1](https://github.com/WandererXII/lishogi/assets/43043181/83d2a58b-d2a5-44b2-abcd-ce3192d9a739)

I kinda like having a check mark in the corner, although it's not immediately obvious what it means.
![alt5](https://github.com/WandererXII/lishogi/assets/43043181/6a44c8e4-b2e1-4661-a661-e732dd54fc80)
Could also be green, but maybe it's too much:
![image](https://github.com/WandererXII/lishogi/assets/43043181/8a07137d-8176-4fc8-bb18-c742af279e28)

**Comment by Olekaze** [comment]
I don't really like autopairing correspondence games. If you like to play 1 day games but someone already has five 1 day game seeks, it's impossible for to create games without first accepting the 5 games from the same person

**Comment by WandererXII** [comment]
That's the current behavior too right? This seems important, I am considering keeping some modified auto pairing when going from the preset - for example only auto pairing you with an opponent if you were not paired with said opponent in the last n games, but no auto pairing whatsoever when creating the game manually.
But maybe that's just confusing... So perhaps no auto pairing at all for correspondence is better.

**Comment by Olekaze** [comment]
I like no autopairing at all, though it may be harder to find games with too many game seeks. you can only see 13 seeks at once and the correspondence lobby has no filter

**Comment by WandererXII** [comment]
Added filter, chart view, sort and separated variants and standard shogi - making it look more like the real time lobby tab.
![image](https://github.com/WandererXII/lishogi/assets/43043181/278aa6ae-a8f7-4948-81ca-981b6496d8af)

I changed it so there will be no auto pairing for correspondence when creating a game manually, and when clicking a preset it will not auto pair you with someone you were already paired with within the last hour.

**Comment by WandererXII** [comment]
I reverted the no auto pairing behavior (https://github.com/WandererXII/lishogi/commit/96043e90d4d222d5ba6c93b44ce609d2a77dc75b), since I thought it might result in too many duplicates and less matches overall, especially if some people came to expect the auto pairing.
But to avoid matching the same players over and over, I made it that you won't be matched with players you are already playing a correspondence game with - which with limited player base is maybe similar to no auto pairing...

**Comment by WandererXII** [comment]
Done - https://lishogi.org/blog/post/ZcQjlRAAACMAgvrF. For issues with the current implementation create a new issue please.
